src/main/java/org/folio/helper/BindHelper.java

```diff
@@ -93,7 +93,7 @@ private Future<Void> removeForbiddenEntities(Piece piece, RequestContext request
 
   private Future<Void> clearTitleBindItemsIfNeeded(String titleId, String bindItemId, RequestContext requestContext) {
     String query = String.format("titleId==%s and bindItemId==%s and isBound==true", titleId, bindItemId);
-    return pieceStorageService.getPieces(0, 0, query, requestContext)
+    return pieceStorageService.getAllPieces(query, requestContext)
```

Comment:
we just need this call to get totalRecords, but we do not need pieces, so in order to not load DB can use
getPieces(0, 0, query, requestContext)

Reply:
Good catch, fixed